app/controllers/split_checkout_controller.rb

```diff
@@ -150,6 +154,31 @@ def update_order
     @order.errors.empty?
   end
 
+  def use_shipping_address_from_distributor
+    @order.ship_address = @order.address_from_distributor
+
+    # Add the missing data
+    bill_address = params[:order][:bill_address_attributes]
+    @order.ship_address.firstname = bill_address[:firstname]
+    @order.ship_address.lastname = bill_address[:lastname]
+    @order.ship_address.phone = bill_address[:phone]
```

Comment:
I can see that `@order.address_from_distributor` will include the name and phone number from billing address already. So I think that will pass for the validation sake.

Of course, this captures any user input that was just changed. But if `Order.updater` is overwriting it later anyway, I'm wondering if it's necessary here? Maybe we can remove this?

Reply:
> Of course, this captures any user input that was just changed. But if `Order.updater` is overwriting it later anyway, I'm wondering if it's necessary here? Maybe we can remove this?

It is necessary because at the time `@order.address_from_distributor` is called annoyingly `order.bill_address` isn't set, so it's needed for `@order.ship_address` validation to pass